web/components/charts/contract/cert.tsx

```diff
@@ -0,0 +1,91 @@
+import { useMemo } from 'react'
+import { last, max, sortBy } from 'lodash'
+import { curveStepAfter } from 'd3-shape'
+
+import { DAY_MS } from 'common/util/time'
+import { CertContract } from 'common/contract'
+import { NUMERIC_GRAPH_COLOR } from 'common/numeric-constants'
+import {
+  formatDateInRange,
+  getRightmostVisibleDate,
+  TooltipProps,
+} from '../helpers'
+import { HistoryPoint, SingleValueHistoryChart } from '../generic-charts'
+import { scaleLinear, scaleTime } from 'd3-scale'
+import { CertTxn } from 'common/txn'
+import { Row } from 'web/components/layout/row'
+
+const MARGIN = { top: 20, right: 40, bottom: 20, left: 10 }
+const MARGIN_X = MARGIN.left + MARGIN.right
+const MARGIN_Y = MARGIN.top + MARGIN.bottom
+
+const CertChartTooltip = (
+  props: TooltipProps<Date, HistoryPoint<Partial<CertTxn>>>
```

Comment:
This and all the other `HistoryPoint<...>` types in this file should just be `HistoryPoint`, right? You don't actually put the `CertTxn` in the point.